super().__init_subclass__() fails

```
$ mypy bin/insubcl.py 
bin/insubcl.py:3: error: Too few arguments for "__init_subclass__" of "object"
$ cat bin/insubcl.py 
class X:
    def __init_subclass__(cls) -> None:
        super().__init_subclass__()
$ mypy --version
mypy 0.730+dev.36fcf1d5aa0f5656ac367bc66aba6e390f82ea9a.dirty
```

This code works correctly at runtime and the latest release accepts it.

--- a/mypy/nodes.py
+++ b/mypy/nodes.py
@@ -439,7 +439,7 @@ class FuncBase(Node):
                  'unanalyzed_type',
                  'info',
                  'is_property',
-                 'is_class',        # Uses "@classmethod"
+                 'is_class',        # Uses "@classmethod" (explicit or implicit)
                  'is_static',       # Uses "@staticmethod"
                  'is_final',        # Uses "@final"
                  '_fullname',

--- a/mypy/semanal.py
+++ b/mypy/semanal.py
@@ -611,10 +611,12 @@ class SemanticAnalyzer(NodeVisitor[None],
             if not func.arguments:
                 self.fail('Method must have at least one argument', func)
             elif isinstance(functype, CallableType):
+                if func.name() == '__init_subclass__':
+                    func.is_class = True
                 self_type = functype.arg_types[0]
                 if isinstance(self_type, AnyType):
                     leading_type = fill_typevars(info)  # type: Type
-                    if func.is_class or func.name() in ('__new__', '__init_subclass__'):
+                    if func.is_class or func.name() == '__new__':
                         leading_type = self.class_type(leading_type)
                     func.type = replace_implicit_first_type(functype, leading_type)
 

--- a/test-data/unit/check-super.test
+++ b/test-data/unit/check-super.test
@@ -254,8 +254,9 @@ def g(b: B) -> None:
 
 [case testSuperInInitSubclass]
 class A:
-    def __init_subclass(cls) -> None:
+    def __init_subclass__(cls) -> None:
         super().__init_subclass__()
+[builtins fixtures/__init_subclass__.pyi]
 
 
 -- Invalid uses of super()

--- a/test-data/unit/check-super.test
+++ b/test-data/unit/check-super.test
@@ -252,6 +252,11 @@ def g(b: B) -> None:
     super(B, b).f('42')
     super(B, b).f(42) # E: Argument 1 to "f" of "A" has incompatible type "int"; expected "str"
 
+[case testSuperInInitSubclass]
+class A:
+    def __init_subclass(cls) -> None:
+        super().__init_subclass__()
+
 
 -- Invalid uses of super()
 -- -----------------------

--- a/test-data/unit/fixtures/__init_subclass__.pyi
+++ b/test-data/unit/fixtures/__init_subclass__.pyi
@@ -1,0 +1,11 @@
+# builtins stub with object.__init_subclass__
+
+class object:
+    def __init_subclass__(cls) -> None: pass
+
+class type: pass
+
+class int: pass
+class bool: pass
+class str: pass
+class function: pass

